R/backend-dbplyr__duckdb_connection.R

```diff
@@ -77,6 +77,18 @@ duckdb_grepl <- function(pattern, x, ignore.case = FALSE, perl = FALSE, fixed =
   }
 }
 
+duckdb_n_distinct <- function(..., na.rm = FALSE) {
+  sql <- pkg_method("sql", "dbplyr")
+
+  if (!identical(na.rm, FALSE)) {
+    stop("Parameter `na.rm = TRUE` in n_distinct() is currently not supported in DuckDB backend.", call. = FALSE)
+  }
+
+  # https://duckdb.org/docs/sql/data_types/struct.html#creating-structs-with-the-row-function
+  str_struct <- paste0("row(", paste0(list(...), collapse = ", "), ")")
+
+  sql(paste0("COUNT(DISTINCT ", str_struct, ")"))
```

Comment:
@mgirlich: Can you please help me review this part of the code?

Reply:
Basically looks good to me. This also works for diverse user input, e.g. `n_distinct(x, a / (b - 1))` or `n_distinct(!!x)`.
One minor thing: you might want to check for empty dots.